.circleci/config.yml

```diff
@@ -1,36 +1,151 @@
+          command: emulator @Pixel_3a_API_26 -no-window -no-audio
+          background: true
+
+      - run:
+          name: wait for emulator
+          command: |
+            adb wait-for-device shell 'while [[ -z $(getprop dev.bootcomplete) ]]; do sleep 1; done;'
+  
+  npm-dependencies:
+    description: "installs all global and local npm dependencies"
+    steps:
+      - run: npm install -g cordova
+      - run: npm install -g github:apache/cordova-paramedic
+      - run: npm install -g ios-deploy
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v1-dependencies-
+      - run: npm install
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+    
+  prepare-tests:
+    description: "prepares tests to run"
+    steps:
+      - run: sed -i .bck s/api_key/$API_KEY/ tests/tests.js
+
+  simulator-dependencies:
+    description: "Install iOS simulator dependencies"
+    steps:
+      - run:
+          name: "Install applesimutils"
+          command: |
+            HOMEBREW_NO_AUTO_UPDATE=1 brew tap wix/brew
+            HOMEBREW_NO_AUTO_UPDATE=1 brew install applesimutils
+  
 jobs:
-  runtest:
+  android-integration-test:
+    description: "Run Android integration tests for Flutter"
     macos:
-      xcode: "10.2.0"
+      xcode: 11.3.1
+    steps:
+      - checkout
+      - run: brew install gradle
+      - android-sdk-dependencies
+      - create-launch-android-emulator
+      - npm-dependencies
+      - prepare-tests
+      - run: npm run test:android
 
+  ios-integration-test:
+    description: "Run iOS integration tests for Flutter"
+    macos:
+      xcode: 11.3.1
     steps:
       - checkout
+      - run:
+          name: Open simulator
+          command: xcrun instruments -w "iPhone 11 Pro (13.3) [" || true
+      - npm-dependencies
+      - prepare-tests
+      - run: npm run test:ios
 
-      # Download and cache dependencies
+  runtest:
+    macos:
+      xcode: 11.3.1
+
+    steps:
+      - checkout
       - restore_cache:
           keys:
             - v1-dependencies-{{ checksum "package.json" }}
-            # fallback to using the latest cache if no exact match is found
             - v1-dependencies-
-
       - run: yarn
-
       - save_cache:
           paths:
             - node_modules
           key: v1-dependencies-{{ checksum "package.json" }}
-
       - run: yarn test
-
       - run: yarn tslint
-
       - run:
           name: Build
           command: |
             yarn build
+
 workflows:
   version: 2
   build-test:
     jobs:
       - runtest
+  
+  android-integration-test:
+    jobs:
+      - android-integration-test:
+          filters:
+            tags:
+              only: /^(\d+\.)(\d+\.)(\d+)$/
+#            Uncomment before merging
+#            branches:
+#              only: /^release\/(\d+\.)(\d+\.)(\d+)$/
+  
+  ios-integration-test:
+    jobs:
```

Comment:
we're not assembling releases or preparing archives with these, if it's intended for future work we should add it to clubhouse